compiler/modules/front.syntax/ast.bal

```diff
@@ -462,7 +462,8 @@ public type ListTypeDesc record {|
     Position startPos;
     Position endPos;
     TypeDesc[] members;
-    TypeDesc rest;
+    TypeDesc? rest;
```

Comment:
We need to have separate ArrayTypeDesc and TupleTypeDesc types. These are too different now.